packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { ATLAS_SEARCH_TEMPLATES } from '@mongodb-js/mongodb-constants';
+import type { SearchTemplate } from '@mongodb-js/mongodb-constants';
+import React, { useState, useCallback } from 'react';
+import {
+  Select,
+  Option,
+  Icon,
+  css,
+  spacing,
+  Tooltip,
+} from '@mongodb-js/compass-components';
+
+const dropdownStyles = css({
+  display: 'flex',
+  overflow: 'hidden',
+  gap: 0,
+});
+
+const overlapDropdownLabelStyles = css({
+  marginLeft: -spacing[3],
+});
+
+type SearchIndexTemplateDropdownProps = {
+  tooltip: string;
+  onTemplate: (template: SearchTemplate) => void;
+};
+
+export const SearchIndexTemplateDropdown: React.FunctionComponent<
+  SearchIndexTemplateDropdownProps
+> = ({ tooltip, onTemplate }) => {
+  const [templateValue, setTemplateValue] = useState('0');
+
+  const onChooseTemplate = useCallback(
+    (value: string) => {
+      setTemplateValue(value);
+      onTemplate(ATLAS_SEARCH_TEMPLATES[+value]);
+    },
+    [onTemplate]
+  );
+
+  return (
+    <section className={dropdownStyles}>
+      <Select
```

Comment:
If your container is of a fixed size of 40%, this should probably stretch to the whole container, otherwise because some of the labels are shorter, some are longer, you get weird empty space on the side:

![Uploading image.png…]()

Reply:
I'll try the suggestion to use a component instead of a string for the label, that should fix it.

Reply:
With a component it works and the type system does not complain, so I'll go that way. Thanks a lot @gribnoysup!

Reply:
Interesting, in their types that I see they only allow strings (but yeah, anyway, the way they handle the prop we can pass components there):

![image](https://github.com/mongodb-js/compass/assets/5036933/cddd3e9f-becd-4904-bd76-6e8098b9a812)

Reply:
> I've checked and it didn't complain for some reason when compiling, but it failed with check-ci, so there is something definitely weird there

That's expected, we don't break the locally running dev build on compilation errors, otherwise you'd have failures all the time while working on a feature locally and this is not productive. This issues are something that for most of us would be highlighted by typescript or eslint plugin in vscode before you catch it by explicitly running check command